accepted/0000-npm-option-handling.md

```diff
@@ -0,0 +1,190 @@
+# Handling of npm config options in the CLI and dep stack
+
+NB: this is an internal RFC related to the code structure and patterns of
+the CLI and its dependencies.  No user-facing changes are contemplated by
+this proposal.
+
+## Summary
+
+- Replace usage of `figgy-pudding` with plain JavaScript option objects.
+- Document and define defaults for configs used by each module in the
+  stack.
+- Clone options objects if they may be mutated in the future.
+- Pass options objects intact to components further down the stack.
+- Canonicalize option names to match consistent `camelCase` format.
+
+## Motivation
+
+The goal of FiggyPudding was to get the npm CLI and its associated
+dependencies away from having a shared `npmconf` god-object with `.get()`
+and `.set()` methods, which could be mutated at any time, and contain any
+arbitrary fields.
+
+However, the use of FiggyPudding has led in practice to added complexity
+and an even _more_ entrenched reliance on a god-object pattern, albeit one
+that is even more opinionated and harder to use than `npmconf` was.
+
+In one example where this has proven particularly challenging, `npm ci`
+does not get the full set of config values, and thus it does not properly
+respect config values that would be handled if the user ran `npm install`.
+Since _all_ package tree generation and reification will be handled by an
+external dependency in npm v7 (ie, `@npmcli/arborist`), this will become an
+even bigger issue.
+
+## Detailed Explanation
+
+In all of the CLI's dependencies, remove `figgy-pudding`, and replace with
+an options object that uses the canonical names for npm config values.
+
+In the npm CLI, create n options object with these canonically named
```

Comment:
Looks like a typo slipped through (ie. `...create "an" options...` unless you mean `n` in terms of *Big O* notation

Reply:
"an" == "n" where `n === 1`.  Which it does in this case.  (I'll fix the typo, thanks :)